cert/storage/disk.go

```diff
@@ -0,0 +1,172 @@
+package storage
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+
+	"github.com/cloudflare/certmgr/cert/storage/util"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+// FileBackend is used for storing PKI content to disk and enforcing permissions.
+type FileBackend struct {
+	ca   *util.CertificateFile
+	cert *util.CertificateFile
+	key  *util.File
+}
+
+// NewFileBackend creates a storage backend used for writing CA/keypair's to disk.
+func NewFileBackend(ca *util.CertificateFile, cert *util.CertificateFile, key *util.File) (*FileBackend, error) {
+	if (cert == nil) != (key == nil) {
+		return nil, errors.New("if either cert or key are defined, both fields must be defined as must request to successfully write the keypair to disk")
```

Comment:
I can't understand what this error message means, and I imagine users wouldn't understand either :) Not sure what to suggst since I don't quite understand... but I think "both cert and key must be specified if either are specified".

Reply:
The english on that one is sub-optimal.  It's supposed to error out if you specify just cert or just key- it requires both.

I'll rephrase it shortly.

Reply:
english is updated.  Resolve if there isn't room for improvement please.

Reply:
SGTM :)